IdentityCore/src/util/NSURL+MSIDExtensions.m

```diff
@@ -133,6 +133,17 @@ - (NSURL *)msidURLForHost:(NSString *)preferredHost context:(id<MSIDRequestConte
         // probably a good thing.
         components.percentEncodedHost = hostComponents[0];
         
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 160000 || __MAC_OS_X_VERSION_MAX_ALLOWED >= 130000
+        if (@available(iOS 16.0, macOS 13.0, *))
+        {
+            // On iOS 16.0 or macOS 13.0 and above, NSURLComponents percentEncodedHost will no longer throw an exception if invalid.
+            if ([NSString msidIsStringNilOrBlank:components.host])
+            {
+                @throw MSIDException(MSIDGenericException, @"Host is not valid.", nil);
```

Comment:
Should we just return error directly here like on line 167?

Reply:
+1

Reply:
Makes sense, updated.